src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -46,406 +47,423 @@
+                                return;
+                            case GRAB_MRDT_PHOTO:
+                                String js = mrdt.process(requestCode, resultCode, i);
+                                trace(this, "Execing JS: %s", js);
+                                evaluateJavascript(js);
+                                return;
+                            default:
+                                trace(this, "onActivityResult() :: no handling for requestCode=%s", requestCode);
+                        }
+                    } else {
+                        String js = simprints.process(requestCode, i);
+                        trace(this, "Execing JS: %s", js);
+                        evaluateJavascript(js);
+                    }
+                } catch(Exception ex) {
+                    String action = i == null ? null : i.getAction();
+                    warn(ex, "Problem handling intent %s (%s) with requestCode=%s & resultCode=%s", i, action, requestCode, resultCode);
+                }
+      }
 
 //> ACCESSORS
-	SimprintsSupport getSimprintsSupport() {
-		return this.simprints;
-	}
+      SimprintsSupport getSimprintsSupport() {
+          return this.simprints;
+      }
 
-	MrdtSupport getMrdtSupport() {
-		return this.mrdt;
-	}
+      MrdtSupport getMrdtSupport() {
+        return this.mrdt;
+      }
 
-	SmsSender getSmsSender() {
-		return this.smsSender;
-	}
+      SmsSender getSmsSender() {
+          return this.smsSender;
+      }
 
 //> PUBLIC API
-	public void evaluateJavascript(final String js) {
-		container.post(new Runnable() {
-			public void run() {
-				// `WebView.loadUrl()` seems to be significantly faster than
-				// `WebView.evaluateJavascript()` on Tecno Y4.  We may find
-				// confusing behaviour on Android 4.4+ when using `loadUrl()`
-				// to run JS, in which case we should switch to the second
-				// block.
-				// On switching to XWalkView, we assume the same applies.
-				if(true) { // NOPMD
-					container.load("javascript:" + js, null);
-				} else {
-					container.evaluateJavascript(js, IGNORE_RESULT);
-				}
-			}
-		});
-	}
-
-	public void errorToJsConsole(String message, Object... extras) {
-		jsConsole("error", message, extras);
-	}
-
-	public void logToJsConsole(String message, Object... extras) {
-		jsConsole("log", message, extras);
-	}
-
-//> PRIVATE HELPERS
-	private void jsConsole(String type, String message, Object... extras) {
-		String formatted = String.format(message, extras);
-		String escaped = formatted.replace("'", "\\'");
-		evaluateJavascript("console." + type + "('" + escaped + "');");
-	}
-
-	private void configureUseragent() {
-		String current = container.getUserAgentString();
-
-		container.setUserAgentString(createUseragentFrom(current));
-	}
-
-	private void openSettings() {
-		startActivity(new Intent(this,
-				SettingsDialogActivity.class));
-		finish();
-	}
-
-	private String getRootUrl() {
-		return appUrl + (DISABLE_APP_URL_VALIDATION ?
```

Comment:
So this part tries to address https://github.com/medic/cht-android/issues/94.

This constant `DISABLE_APP_URL_VALIDATION` can be set through an environment variable with the same name when building the app, but the default value when the variable is not set is `false`, making the default behavior to add to the CHT url the `"/medic/_design/medic/_rewrite/"` string at the end.

We can either:
- Remove this behavior as you did. In that case we have to remove any other occurrence of the variable `DISABLE_APP_URL_VALIDATION` that is used here to change the URL, but it's also referenced in the `Makefile`, the `build.gradle` where is defined and lastly in the class `AppUrlVerifier`, where is actually used to disable the checks if it's `true`, but doesn't make sense to disable the URL validations, so I would remove [this block](https://github.com/medic/cht-android/blob/f9bc1e1b1b94d1a8cf9f0cfb177f412db0471bbc/src/main/java/org/medicmobile/webapp/mobile/AppUrlVerifier.java#L18-L20) too:
  ```java
		if(DISABLE_APP_URL_VALIDATION) {
			return AppUrlVerification.ok(appUrl);
		}
  ```
- Change the default behavior, making `true` the default value for the constant `DISABLE_APP_URL_VALIDATION` , or changing the constant to a new one with the opposite effect, e.g. `ENABLE_APP_URL_VALIDATION`, either way the name of the variable underrepresented what it does, because not just enable / disable the validation of the URL but also overwrites its value.

Taking into account that the overriding of the URL is an old behavior used in old versions of the CHT that are not supported anymore and almost incompatible with the latest versions of the Android app, I would remove the overriding as this PR does, remove any occurrence of the variable `DISABLE_APP_URL_VALIDATION`, and always validate the URL anyway.

@garethbowen , what do you think?

Reply:
I agree with removing it. To be extra safe we could put off this change until v1.0.0 which will document the potential breaking change and have a minimum dependency on Core v4.0.0. What do you think?